kmath-nd4j/src/main/kotlin/kscience.kmath.nd4j/INDArrayAlgebra.kt

```diff
@@ -0,0 +1,284 @@
+package kscience.kmath.nd4j
+
+import org.nd4j.linalg.api.ndarray.INDArray
+import org.nd4j.linalg.factory.Nd4j
+import kscience.kmath.operations.*
+import kscience.kmath.structures.*
+
+/**
+ * Represents [NDAlgebra] over [INDArrayAlgebra].
+ *
+ * @param T the type of ND-structure element.
+ * @param C the type of the element context.
+ */
+public interface INDArrayAlgebra<T, C> : NDAlgebra<T, C, INDArrayStructure<T>> {
```

Comment:
I do not like naming. Could we use `Nd4jArrayAlgebra` etc?

Reply:
Yes

Reply:
It is 5 minutes to refactor.